lib/galaxy/jobs/search.py

```diff
@@ -0,0 +1,143 @@
+import json
+from six import string_types
+from sqlalchemy import and_, or_
+from sqlalchemy.orm import aliased
+
+from galaxy import model
+from galaxy.managers.hdas import HDAManager
+from galaxy.managers.collections import DatasetCollectionManager
+from galaxy.managers.lddas import LDDAManager
+
+
+class JobSearch(object):
```

Comment:
I'd not put this file in ``galaxy.jobs``, I think you are implementing something that belongs in ``galaxy.managers`` somewhere - probably ``galaxy.managers.jobs``. ``galaxy.jobs`` to me feels more like a low level thing related to dealing with actually running jobs. This is implementing manager-style business logic.